pkg/dockerregistry/server/auth.go

```diff
@@ -343,12 +344,36 @@ func (ac *AccessController) Authorized(ctx context.Context, accessRecords ...reg
 				}
 				verifiedPrune = true
 			default:
+				if repoAccessVerified {
```

Comment:
right and it is getting more complicated this way... I think we can just check the access twice and don't get too smart when trying to optimize this.